MantidQt/CustomInterfaces/src/SANSRunWindow.cpp

```diff
@@ -1702,18 +1700,21 @@ void SANSRunWindow::setGeometryDetails() {
     return;
   }
 
-  const auto &dets = monitorWs->getSpectrum(monitorWsIndex).getDetectorIDs();
-  if (dets.empty())
+  const auto &monitorDetectorIDs =
+      monitorWs->getSpectrum(monitorWsIndex).getDetectorIDs();
+  if (monitorDetectorIDs.empty())
     return;
 
   double dist_mm(0.0);
   QString colour("black");
-  try {
-    Mantid::Geometry::IDetector_const_sptr detector =
-        instr->getDetector(*dets.begin());
 
-    double unit_conv(1000.);
-    dist_mm = detector->getDistance(*source) * unit_conv;
+  const auto &spectrumInfo = sampleWs->spectrumInfo();
+
+  try {
+    const auto &detector = spectrumInfo.detector(*monitorDetectorIDs.begin());
+    const double unit_conv(1000.);
+    const auto &source = sampleWs->getInstrument()->getSource();
+    dist_mm = detector.getDistance(*source) * unit_conv;
```

Comment:
@DavidFair 
The original code was `instr->getDetector(*dets.begin())` and not `sampleWs->getDetector(...)` which then suggests the usage of `DetectorInfo` rather than `SpectrumInfo`.